src/QtAVPlayer/qavaudiooutput.cpp

```diff
@@ -147,7 +147,7 @@ class QAVAudioOutputPrivate : public QIODevice
         if (!audioOutput || (fmt.isValid() && audioOutput->format() != fmt) || audioOutput->state() == QAudio::StoppedState) {
             if (audioOutput)
                 audioOutput->deleteLater();
-            audioOutput = new AudioOutput(fmt);
+            audioOutput = new AudioOutput(QAudioDevice(QMediaDevices::defaultAudioOutput()),fmt);
```

Comment:
This is not compatible with Qt5

Reply:
Does it mean that default audio device is not used?  But should it be used even if it is not provided?

Reply:
I've implemented a simple video player and have noticed the following. In case there is more than one audio device connected to the computer QtAVPlayer cannot switch to the default device automatically. I.e. when I run the player with my embedded speakers on and I connect my headphones I hear nothing. My suggestion is to add an explicit reference to the default audio output. Tested on Manjaro Linux.

Reply:
I think QtAVPlayer should switch to the default audio device every time it changes.

Reply:
Could you try this? If it works, could you just edit the PR by this change?
```
diff --git a/src/QtAVPlayer/qavaudiooutput.cpp b/src/QtAVPlayer/qavaudiooutput.cpp
index 7ced332..355140b 100644
--- a/src/QtAVPlayer/qavaudiooutput.cpp
+++ b/src/QtAVPlayer/qavaudiooutput.cpp
@@ -17,6 +17,7 @@
 #include <QAudioOutput>
 #else
 #include <QAudioSink>
+#include <QMediaDevices>
 #endif
 
 extern "C" {
@@ -89,8 +90,10 @@ public:
 
 #if QT_VERSION < QT_VERSION_CHECK(6, 0, 0)
     using AudioOutput = QAudioOutput;
+    using AudioDevice = QAudioDeviceInfo;
 #else
     using AudioOutput = QAudioSink;
+    using AudioDevice = QAudioDevice;
 #endif
     AudioOutput *audioOutput = nullptr;
     qreal volume = 1.0;
@@ -98,6 +101,7 @@ public:
     QList<QAVAudioFrame> frames;
     qint64 offset = 0;
     bool quit = 0;
+    AudioDevice defaultAudioDevice;
     mutable QMutex mutex;
     QWaitCondition cond;
     QThreadPool threadPool;
@@ -144,10 +148,21 @@ public:
 
     void init(const QAudioFormat &fmt)
     {
-        if (!audioOutput || (fmt.isValid() && audioOutput->format() != fmt) || audioOutput->state() == QAudio::StoppedState) {
+#if QT_VERSION < QT_VERSION_CHECK(6, 0, 0)
+        auto audioDevice = QAudioDeviceInfo::defaultOutputDevice();
+#else
+        auto audioDevice = QMediaDevices::defaultAudioOutput();
+#endif
+        if (!audioOutput
+            || (fmt.isValid() && audioOutput->format() != fmt)
+            || audioOutput->state() == QAudio::StoppedState
+            || defaultAudioDevice != audioDevice)
+        {
             if (audioOutput)
                 audioOutput->deleteLater();
-            audioOutput = new AudioOutput(fmt);
+            audioOutput = new AudioOutput(audioDevice, fmt);
+            defaultAudioDevice = audioDevice;
+
             QObject::connect(audioOutput, &AudioOutput::stateChanged, audioOutput,
                 [&](QAudio::State state) {
                     switch (state) {
```

Reply:
I tested this by 
1. Starting playing
2. Unplug headset
3. Plug the headset back --- should hear sound.